### PR TITLE
[ci] Bring up LUCI Linux custom package tests

### PR DESCRIPTION
**Pull request by stuartmorgan**
Adds LUCI version of Linux custom package tests.

Removes the custom tests for `cross_file` and `google_identity_services_web` since they only existed to run unit tests in browser mode, which is [now standard for all packages that can be run that way](https://github.com/flutter/packages/pull/4378).

Part of https://github.com/flutter/flutter/issues/114373

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -97,7 +97,8 @@ targets:
     recipe: packages/packages
     timeout: 60
     properties:
-      target_file: dart_unit_tests.yaml
+      # DO NOT LAND - for presubmit validation only.
```
Comment (stuartmorgan) [inline .ci.yaml:100]: This is a WIP so I can find any issues in presubmit, and I'll revert it once that's done. The rest is real code though.

File: .ci.yaml [hunk 2]
```diff
@@ -214,6 +214,40 @@ targets:
       channel: stable
       version_file: flutter_stable.version
 
+  - name: Linux_android custom_package_tests master
```
Comment (ditman) [inline .ci.yaml:217]: Does the name of these need to start with `Linux_android`? Why not just `Linux custom_package_tests master`?

File: .ci.yaml [hunk 3]
```diff
@@ -214,6 +214,40 @@ targets:
...
+    bringup: true # New target
+    recipe: packages/packages
+    timeout: 30
+    properties:
+      add_recipes_cq: "true"
+      version_file: flutter_master.version
+      target_file: linux_custom_package_tests.yaml
+      cores: "32"
+      # Pigeon tests need Andoid deps (thus the Linux_android base) and
```
Comment (ditman) [inline .ci.yaml:226]:
```suggestion
      # Pigeon tests need Android deps (thus the Linux_android base) and
```

Is this sufficient to name the suite "Linux_android" though? :P

File: .ci.yaml [hunk 4]
```diff
@@ -214,6 +214,40 @@ targets:
...
+  - name: Linux_android custom_package_tests stable
+    bringup: true # New target
+    recipe: packages/packages
+    timeout: 30
+    properties:
+      add_recipes_cq: "true"
+      version_file: flutter_stable.version
+      target_file: linux_custom_package_tests.yaml
+      cores: "32"
+      # Pigeon tests need Andoid deps (thus the Linux_android base) and
```
Comment (ditman) [inline .ci.yaml:243]:
```suggestion
      # Pigeon tests need Android deps (thus the Linux_android base) and
```

File: .ci.yaml (hunk 3 above)
Comment (tarrinneal) [reply .ci.yaml:226]: I like andoid better

**Review by tarrinneal — APPROVED** [review]
Approval from me once you and @ditman resolve your differences.

File: .ci.yaml (hunk 3 above)
Comment (ditman) [reply .ci.yaml:226]: `Linux_andoid custom_package_tests master` :P

File: .ci.yaml (hunk 2 above)
Comment (stuartmorgan) [reply .ci.yaml:217]: That would use the more generic `Linux` configuration, and then I would have to duplicate all the dependencies in each of the targets. I considered it, but opted for less repetition over preferred naming.

The rigidity of LUCI Cindy naming and how it determines behavior is definitely not one of my favorite parts of LUCI. I would much rather the base config be a property unrelated to name.

**Pull request merged**